Src/IronPython/Runtime/PythonRange.cs

```diff
@@ -5,115 +5,132 @@
 // Copyright (c) Pawel Jasinski.
 //
 
+#nullable enable
+
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Numerics;
 
 using IronPython.Runtime.Binding;
 using IronPython.Runtime.Operations;
 using IronPython.Runtime.Types;
 
 using Microsoft.Scripting.Runtime;
+using Microsoft.Scripting.Utils;
+
+using NotNullAttribute = Microsoft.Scripting.Runtime.NotNullAttribute;
 
 namespace IronPython.Runtime {
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix")]
     [PythonType("range")]
     [DontMapIEnumerableToContains]
-    public sealed class PythonRange : IEnumerable<int>, ICodeFormattable, IList, IReversible {
-        private int _length;
-
-        public PythonRange(object stop) : this(0, stop, 1) { }
-        public PythonRange(object start, object stop) : this(start, stop, 1) { }
-
-        public PythonRange(object start, object stop, object step) {
-            Initialize(start, stop, step);
-        }
-
-        private void Initialize(object ostart, object ostop, object ostep) {
-            // TODO: https://github.com/IronLanguages/ironpython3/issues/472 - should not throw but it's better than an incorrect result...
-            stop = Converter.ConvertToIndex(ostop, throwOverflowError: true);
-            start = Converter.ConvertToIndex(ostart, throwOverflowError: true);
-            step = Converter.ConvertToIndex(ostep, throwOverflowError: true);
-            if (step == 0) {
+    public sealed class PythonRange : ICodeFormattable, IReversible, IEnumerable<int>, IEnumerable<BigInteger> {
+        internal readonly BigInteger _start;
+        internal readonly BigInteger _stop;
+        internal readonly BigInteger _step;
+        internal readonly BigInteger _length;
+
+        public PythonRange([AllowNull] object stop) : this(0, stop, 1) { }
```

Comment:
I didn't want to use `object?` since `null` is expected to throw, but I didn't want to use `[NotNullDynamic] object` since I still wanted it to bind and throw with the standard indexing error instead of the normal `TypeError: expected object, got NoneType`. However I just tried changing the constructor to `public PythonRange([NotNull] object stop)` and `range(None)` still throwing the index error so I'm confused. 😕